packages/hierarchy-builder/src/test/internal/operators/grouping/PropertiesGrouping.test.ts

```diff
@@ -0,0 +1,1482 @@
+          createTestProcessedInstanceNode({
+            key: { type: "instances", instanceKeys: [{ className: "TestSchema.Class", id: "0x1" }] },
+            processingParams: {
+              grouping: {
+                byProperties: {
+                  fullClassName: "TestSchema.Class",
+                  propertyGroups: [{ propertyName: "PropertyName", propertyValue: 5, ranges: [{ fromValue: 1, toValue: 5, rangeLabel: "rangeLabel" }] }],
+                },
+              },
+            },
+          }),
+          createTestProcessedInstanceNode({
+            key: { type: "instances", instanceKeys: [{ className: "TestSchema.Class", id: "0x2" }] },
+            processingParams: {
+              grouping: {
+                byProperties: {
+                  fullClassName: "TestSchema.Class",
+                  propertyGroups: [{ propertyName: "PropertyName", propertyValue: 5, ranges: [{ fromValue: 1, toValue: 5, rangeLabel: "rangeLabel" }] }],
+                },
+              },
+            },
+          }),
+        ];
+        const ecClass = { fullName: "TestSchema.Class" } as unknown as ECClass;
+        classStubs.stubEntityClass({
+          schemaName: "TestSchema",
+          className: "Class",
+          is: async () => true,
+        });
+        const propertyInfo: propertiesGrouping.PropertyGroupInfo = {
+          ecClass,
+          previousPropertiesGroupingInfo: [],
+          propertyGroup: { propertyName: "PropertyName", ranges: [{ fromValue: 1, toValue: 5, rangeLabel: "rangeLabel" }] },
+        };
+        const expectedGroupingNodeKey: GroupingNodeKey = {
+          type: "property-grouping:range",
+          propertyName: "PropertyName",
+          fullClassName: "TestSchema.Class",
+          fromValue: 1,
+          toValue: 5,
+        };
+        expect(await propertiesGrouping.createPropertyGroups(metadataProvider, nodes, propertyInfo, formatter)).to.deep.eq({
+          groupingType: "property",
+          grouped: [
+            createTestProcessedGroupingNode({
+              label: "rangeLabel",
+              key: expectedGroupingNodeKey,
+              children: nodes.map((n) => ({ ...n, parentKeys: [...n.parentKeys, expectedGroupingNodeKey] })),
+            }),
+          ],
+          ungrouped: [],
+        });
+      });
+
+      it("groups when provided range order is different from nodes' ranged property order", async () => {
+        const nodes = [
+          createTestProcessedInstanceNode({
+            key: { type: "instances", instanceKeys: [{ className: "TestSchema.Class", id: "0x1" }] },
+            processingParams: {
+              grouping: {
+                byProperties: {
+                  fullClassName: "TestSchema.Class",
+                  propertyGroups: [
+                    {
+                      propertyName: "PropertyName",
+                      propertyValue: 6,
+                      ranges: [
+                        { fromValue: 1, toValue: 4 },
```

Comment:
shouldn't both ranges accomodate the value in this test?

Reply:
We are looking at the order that is provided in node's grouping params, not the `PropertyGroupInfo`. `PropertyGroupInfo` is only used to identify if `propertyName`, `ECClass` and `ranges` (order can differ from nodes grouping params) match and if they do, that means it is "time" to group the node by its own grouping params.